opentelemetry-sdk/tests/trace/export/test_export.py

```diff
@@ -513,7 +513,7 @@ def test_batch_span_processor_reset_timeout(self):
 
             # give some time for exporter to loop
             # since wait is mocked it should return immediately
-            time.sleep(0.05)
```

Comment:
The test should be marked as [flaky](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/tests/metrics/test_periodic_exporting_metric_reader.py#L193) instead. That way, we actually show that this test is flaky instead of just having a larger sleep.

Reply:
In general I agree, but here we are adding the race condition on purpose by mocking the wait condition in order to test mock asserts so before marking it as flaky I would like to test if adding more slack in timing is helpful

Reply:
All right, if you prefer so :v: